src/main/java/com/microsoft/azure/vmagent/AzureVMCloudPoolRetentionStrategy.java

```diff
@@ -8,33 +8,52 @@
 import hudson.model.Computer;
 import hudson.model.Descriptor;
 import hudson.model.DescriptorVisibilityFilter;
+import hudson.model.Executor;
+import hudson.model.ExecutorListener;
+import hudson.model.Label;
+import hudson.model.Queue;
 import hudson.slaves.Cloud;
 import hudson.slaves.RetentionStrategy;
 import jenkins.model.Jenkins;
 import org.kohsuke.stapler.DataBoundConstructor;
 
 import java.util.Arrays;
 import java.util.List;
+import java.util.Objects;
 import java.util.concurrent.TimeUnit;
 import java.util.logging.Level;
 import java.util.logging.Logger;
+import java.util.stream.Stream;
 
-public class AzureVMCloudPoolRetentionStrategy extends AzureVMCloudBaseRetentionStrategy {
+import javax.annotation.Nonnull;
+
+public class AzureVMCloudPoolRetentionStrategy extends AzureVMCloudBaseRetentionStrategy implements ExecutorListener {
     private static final long serialVersionUID = 1577788691L;
 
     private final long retentionMillis;
 
     private final int poolSize;
 
+    private final boolean sparePool;
+
+    private final boolean singleUse;
+
     private static final long IDLE_LIMIT_MILLIS = TimeUnit.MINUTES.toMillis(1);
 
     private static final Logger LOGGER = Logger.getLogger(AzureVMManagementServiceDelegate.class.getName());
 
     @DataBoundConstructor
-    public AzureVMCloudPoolRetentionStrategy(int retentionInHours, int poolSize) {
+    public AzureVMCloudPoolRetentionStrategy(int retentionInHours, int poolSize, boolean sparePool, boolean singleUse) {
```

Comment:
this is a breaking change for for configuration-as-code users, as they will now be expected to specify these two options, unless these options do need to always be mandatory I would recommend moving them to a `DataBoundSetter` as documented here:
https://jenkins.io/doc/developer/plugin-development/pipeline-integration/#constructor-vs-setters

Reply:
Interesting, I'll look into that, thanks.